BenmoussaYounes/chat-app
# 1-feature add error handling ✨

## JiraTicket [FI-11](https://contributors.atlassian.net/jira/software/projects/ATX/boards/6?selectedIssue=ATX-11)

## Description 📑
- add error handling including snackbar  messages and error screen on errors

## Type of Change


## Changes

This pull request includes the following changes:

- add api error handler to handle errors
- add api error model
- show snackbar on failure api calls
- show error screen with refetch ability on fetching error
- use restartable transformer to fix navigation between screen bug
- add refresh on chat page

## Screenshots 📷

| 01                                    | 02                                    | 
| ------------------------------------- | ------------------------------------- |
| <img src="https://github.com/user-attachments/assets/24d78e62-c82c-499a-a478-e2c3db1b544b" width="100" height="250"> | <img src="https://github.com/user-attachments/assets/ccbcbb86-3421-4e64-8d29-4f57419b0aa5" width="100" height="250"> |

## How to test 🚦
- for api  post calls you can try to send a message with the interent off and a snackbar of failure should appear 
- for streams and api get calls you can include error on the fromJson method for reading the data for example and the error screen should appear with the refetch option
- try to scroll to refresh on chat page to check if the page is refreshing
